client/src/search.tsx

```diff
@@ -89,7 +90,14 @@ function useSearchIndex(): readonly [
           }),
       ];
 
-      const flex = mixed.map(({ title }, i) => [i, title.toLowerCase()]);
+      const flex = mixed.map(
+        ({ title, url }, i) =>
+          [i, title.toLowerCase(), url.split("/").pop().toLowerCase()] as [
+            number,
+            string,
+            string
+          ]
```

Comment:
Avoid `as` as much as possible. We are already TS 4.9, so use `satisfies`?

```suggestion
        ({ title, url }, i) =>
          [i, title.toLowerCase(), url.split("/").pop().toLowerCase()] satisfies [
            number,
            string,
            string
          ]
```

Alternatively, annotate the return type (note, it's better to avoid duplicating the tuple def by using indexed types, or promoting `[index: number, title: string, slugTail: string]` to a separate type def:

```suggestion
        ({ title, url }, i): SearchIndex["flex"][number] =>
          [i, title.toLowerCase(), url.split("/").pop().toLowerCase()]
```

Reply:
TIL about `satisfies`, thanks. I will extract a type def and use it as return type.